refactor: Remove `table.to_csv()` method and add more export options to samples

holtskinner opened the pull request:
- Added CSV, HTML, Markdown output for table in Sample code

holtskinner committed:
refactor: Remove `table.to_csv()` dedicated method and add more export options to samples

- Added CSV, HTML, Markdown output for table in Sample code

gcf-owl-bot[bot] committed:
🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

holtskinner committed:
deps: Add `tabulate` to installation

holtskinner committed:
Merge branch 'main' into tables

holtskinner committed:
fix: Adjust sample to print dataframe

ilai-deutel commented on samples/snippets/table_sample.py:
```diff
@@ -30,11 +30,24 @@ def table_sample(document_path: str, output_file_prefix: str) -> None:
+
+            output_filename = f"{output_file_prefix}-{page.page_number}-{table_index}"
+
+            # Write Dataframe to CSV file
+            with open(f"{output_filename}.csv", "w", encoding="utf-8") as f:
+                f.write(df.to_csv(index=False))
+
+            # Write Dataframe to HTML file
+            with open(f"{output_filename}.html", "w", encoding="utf-8") as f:
+                f.write(df.to_html())
+
+            # Write Dataframe to Markdown file
+            with open(f"{output_filename}.md", "w", encoding="utf-8") as f:
+                f.write(df.to_markdown())
```
Optional suggestions:

- `df.to_*()` functions can take a path instead of a file object as a first argument, it simplifies the code a bit
- I think that `to_html` and `to_markdown` also need `index=False` to avoid outputting a meaningless numerical index column

```suggestion
            # Write Dataframe to CSV file
            df.to_csv(f"{output_filename}.csv", index=False))

            # Write Dataframe to HTML file
            df.to_html(f"{output_filename}.html", index=False)

            # Write Dataframe to Markdown file
            df.to_markdown(f"{output_filename}.md", index=False)
```

holtskinner committed:
docs(samples): Change table sample to use file path for `df._to_*()`

holtskinner committed:
fix: Fixed Lint error

- This wasn't an issue before, I think that `flake8` might have started checking for new criteria.

holtskinner committed:
Merge branch 'main' into tables

holtskinner committed:
Merge branch 'main' into tables

holtskinner committed:
Merge branch 'main' into tables

holtskinner committed:
Merge branch 'main' into tables
